tests/sparseml/transformers/utils/test_helpers.py

```diff
@@ -170,26 +170,43 @@ def test_save_zoo_directory(tmp_path, stub):
 
 
 @pytest.mark.parametrize(
-    "string, prompt, censor, expected_mask",
+    "string, response, prompt, expected_mask",
     [
-        ("[foo]hello\n\n[bar]world", "[foo]", "[bar]", "1111111111110000000000"),
+        (
+            ("[foo]hello\n\n" "[bar]world"),
+            "[bar]",
+            "[foo]",
+            ("000000000000" "1111111111"),
+        ),
         (
             (
                 "[Instruction]python is\n\n"  # 24
                 "[Response]great\n\n"  # 17
                 "[Instruction]What about Java"  # 28
                 "[Response]Meh"  # 13
             ),
-            "[Instruction]",
             "[Response]",
+            "[Instruction]",
             (
-                "111111111111111111111111"  # 24
-                "00000000000000000"  # 17
-                "1111111111111111111111111111"  # 28
-                "0000000000000"  # 13
+                "000000000000000000000000"  # 24
+                "11111111111111111"  # 17
+                "0000000000000000000000000000"  # 28
+                "1111111111111"  # 13
             ),
         ),
+        (
+            ("[foo]hello\n\n" "[bar]world"),
+            "[bar]",
+            None,
+            ("000000000000" "1111111111"),
+        ),
+        (
+            ("hello\n\n" "[bar]world"),
+            "[bar]",
+            None,
+            ("0000000" "1111111111"),
+        ),
     ],
 )
-def test_generate_mask(string, prompt, censor, expected_mask):
-    assert generate_mask(string, prompt, censor) == expected_mask
+def test_generate_mask(string, response, prompt, expected_mask):
```

Comment:
Could we also an in a more E2E test case? Ie loading, processing and tokenizing a few rows of a dataset